core/arch/arm/plat-zynqmp/drivers/puf.h

```diff
@@ -0,0 +1,13 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (C) Foundries Ltd. 2021 - All Rights Reserved
+ * Author: Jorge Ramirez <jorge@foundries.io>
+ */
+
+#ifndef PUF_H_
+#define PUF_H_
+
+TEE_Result puf_regenerate(void);
```

Comment:
There could be other providers for "puf" too... so better to namespace these a bit more.

Reply:
I'll also rename the file

Reply:
so aes -> csu_aes and puf-> csu_pfu and associated interfaces

Reply:
There are files like stm32_ , imx_ ls_, bcm_... so should we have xilinx_mpsoc_X?

There is also folder imx -- one option would be to make core/drivers/xilinx/mpsoc/

We have something coming for zynq so those could then go for core/drivers/xilinx/zynq folder.

@jenswi-linaro do you have comment on how this drivers folder should be organized?

Reply:
Until we see that it doesn't scale I'd suggest using a `zynq_` and `zynqmp_` prefix as needed for the relevant files directly in `core/drivers/`. There's no need to add an additional `xilinx_` prefix.

Drivers below `core/drivers/` are probably better grouped by type than by the name of the soc. It could for instance make sense to put all the uart drivers into `core/drivers/uart/` the day that the number of uart drivers becomes to large.